### PR TITLE
Migrating backplane-cli login & logout

### PR DESCRIPTION
**Pull request by reetika-vyas**
### What type of PR is this?

feature

### What this PR does / Why we need it?

This PR is for migrating backplane login/logout 

### Which Jira/Github issue(s) does this PR fix?

[OSD-14683](https://issues.redhat.com/browse/OSD-14683)

### Pre-checks (if applicable)

- [ Yes] Ran unit tests locally
- [ Yes] Validated the changes in a cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

File: pkg/utils/cluster_test.go [hunk 1]
```diff
@@ -78,10 +78,10 @@ hello: world
 func writeKubeconfigYaml(s string) error {
 	kubeconfigPath := clientcmd.NewDefaultPathOptions().GlobalFile
 	dirname := filepath.Dir(kubeconfigPath)
-	err := os.MkdirAll(dirname, 0744)
-	if err != nil {
-		return err
-	}
+	if _, err := os.Stat(dirname); os.IsNotExist(err) {
+		os.MkdirAll(dirname, 0777)
+		os.Chmod(dirname, 0777)
```
Comment (MitaliBhalla) [inline pkg/utils/cluster_test.go:83]: You will have to handle error properly here, i.e to return it 
```suggestion
		os.Chmod(dirname, 0777)
		if err!=nil{
			return err
		}
```

File: pkg/utils/cluster_test.go [hunk 2]
```diff
@@ -78,10 +78,10 @@ hello: world
 func writeKubeconfigYaml(s string) error {
 	kubeconfigPath := clientcmd.NewDefaultPathOptions().GlobalFile
 	dirname := filepath.Dir(kubeconfigPath)
-	err := os.MkdirAll(dirname, 0744)
-	if err != nil {
-		return err
-	}
+	if _, err := os.Stat(dirname); os.IsNotExist(err) {
+		os.MkdirAll(dirname, 0777)
```
Comment (MitaliBhalla) [inline pkg/utils/cluster_test.go:82]:
```suggestion
		os.MkdirAll(dirname, 0644)
		os.Chmod(dirname, 0777)
```

File: cmd/ocm-backplane/login/login.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+}
+
+func runLogin(cmd *cobra.Command, argv []string) (err error) {
+	var clusterKey string
+	
+	if err != nil {
+		return err
+	}
+
+		// ========= Get The cluster ID =============================
```
Comment (wanghaoran1988) [inline cmd/ocm-backplane/login/login.go:58]:
```suggestion
		//  Get The cluster ID

File: cmd/ocm-backplane/logout/logout_test.go [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+	"github.com/openshift/backplane-cli/pkg/utils"
+)
+
+const (
+	loggedInYamlSingle = `
+apiVersion: v1
+clusters:
+- cluster:
+    server: https://api-backplane.apps.hivep01ue1.b6s7.p1.openshiftapps.com/backplane/cluster/1f0o1maej9brj6j9k6ehbe7rm0k2lng7/
```
Comment (wanghaoran1988) [inline cmd/ocm-backplane/logout/logout_test.go:20]: Please change this url example

**Comment by reetika-vyas** [comment]
Closing this PR due conflicts with branched that merged , due to this tests are failing creating new one

**Pull request closed without merging**